pkg/authn/claims.go

```diff
@@ -0,0 +1,41 @@
+package authn
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	// XMTPD_COMPATIBLE_VERSION_CONSTRAINT major or minor serverVersion bumps indicate backwards incompatible changes
+	XMTPD_COMPATIBLE_VERSION_CONSTRAINT = "~ 0.1.3"
+)
+
+type XmtpdClaims struct {
+	Version *semver.Version `json:"version,omitempty"`
+	jwt.RegisteredClaims
+}
+
+func ValidateVersionClaimIsCompatible(claims *XmtpdClaims) error {
+	if claims.Version == nil {
+		return nil
+	}
```

Comment:
Can you help me understand under which circumstances having a claim with no version is valid? Testing, backwards compatibility?